htdocs/web_portal/views/scopes.php

```diff
@@ -1,6 +1,75 @@
 <div class="rightPageContainer">
-    <h1>Resource Scoping</h1>
-    <br />
+
+    <div style="float: left;">
+        <h1 style="float: left; margin-left: 0em;">
+            Scopes
+        </h1>
+        <span style="clear: both; float: left; padding-bottom: 0.4em;">
+            Click on the name of a scope to <?=$params['UserIsAdmin'] ? "edit it and " : "";?> view objects with that scope tag.
+        </span>
+    </div>
+    <!-- hide add when read only or user is not admin -->
+    <?php if(!$params['portalIsReadOnly'] && $params['UserIsAdmin']):?>
+        <div style="float: right;">
+            <center>
+                <a href="index.php?Page_Type=Admin_Add_Scope">
+                <img src="<?php echo \GocContextPath::getPath()?>img/add.png" height="25px" />
+                <br />
+                <span>Add Scope</span>
+                </a>
+            </center>
+        </div>
+    <?php endif; ?>
+
+    <?php $numberOfScopes = sizeof($params['Scopes'])?>
+    <div class="listContainer">
+        <span class="header listHeader">
+            <?php echo $numberOfScopes ?> Scope<?php if($numberOfScopes) echo "s"?>
+        </span>
+        <table class="vSiteResults" id="selectedSETable">
+            <tr class="site_table_row_1">
+                <th class="site_table">Name</th>
+                <th class="site_table">Description</th>
+                <th class="site_table">Reserved?</th>
+                <?php if(!$params['portalIsReadOnly'] && $params['UserIsAdmin']):?>
+                    <th class="site_table" style="width: 10%">Remove</th>
+                <?php endif; ?>
+            </tr>
+            <?php
+            $num = 2;
+            if(sizeof($numberOfScopes > 0)) {
```

Comment:
Surely ?
```suggestion
            if($numberOfScopes > 0) {
```

Reply:
I would think so, but does that line actually do anything (I didn't write it)? Doesn't the foreach that follows mean that if there are no scopes, no new HTML is added anyway?

Reply:
Yes, looks like it. Imho having it improves readability but it might be better put to use to stop the table header being displayed when there is no content. But that might be more of a change than is wanted here. Minimum I'd say is the 'if' clause should be corrected (even if it always resolves to true, or deleted.